backend/pikpak/pikpak.go

```diff
@@ -0,0 +1,1749 @@
+			opt.Advanced = true
+		} else if opt.Name == config.ConfigClientSecret {
+			opt.Advanced = true
+		}
+		opts = append(opts, opt)
+	}
+	return opts
+}
+
+// pikpakAutorize retrieves OAuth token using user/pass and save it to rclone.conf
+func pikpakAuthorize(ctx context.Context, opt *Options, name string, m configmap.Mapper) error {
+	pass := obscure.MustReveal(opt.Password)
+	t, err := oauthConfig.PasswordCredentialsToken(ctx, opt.Username, pass)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve token using username/password: %w", err)
+	}
+	return oauthutil.PutToken(name, m, t, false)
+}
+
+// Register with Fs
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "pikpak",
+		Description: "PikPak",
+		NewFs:       NewFs,
+		CommandHelp: commandHelp,
+		Config: func(ctx context.Context, name string, m configmap.Mapper, config fs.ConfigIn) (*fs.ConfigOut, error) {
+			// Parse config into Options struct
+			opt := new(Options)
+			err := configstruct.Set(m, opt)
+			if err != nil {
+				return nil, fmt.Errorf("couldn't parse config into struct: %w", err)
+			}
+
+			switch config.State {
+			case "":
+				// Check token exists
+				if _, err := oauthutil.GetToken(name, m); err != nil {
+					return fs.ConfigGoto("authorize")
+				}
+				return fs.ConfigConfirm("authorize_ok", false, "consent_to_authorize", "Re-authorize for new token?")
+			case "authorize_ok":
+				if config.Result == "false" {
+					return nil, nil
+				}
+				return fs.ConfigGoto("authorize")
+			case "authorize":
+				if err := pikpakAuthorize(ctx, opt, name, m); err != nil {
+					return nil, err
+				}
+				return nil, nil
+			}
+			return nil, fmt.Errorf("unknown state %q", config.State)
+		},
+		Options: append(pikpakOAuthOptions(), []fs.Option{{
+			Name:     "user",
+			Help:     "Pikpak username.",
+			Required: true,
+		}, {
+			Name:       "pass",
+			Help:       "Pikpak password.",
+			Required:   true,
+			IsPassword: true,
+		}, {
+			Name: "root_folder_id",
+			Help: `ID of the root folder.
+Leave blank normally.
+
+Fill in for rclone to use a non root folder as its starting point.
+`,
+			Advanced: true,
+		}, {
+			Name:     "use_trash",
+			Default:  true,
+			Help:     "Send files to the trash instead of deleting permanently.\n\nDefaults to true, namely sending files to the trash.\nUse `--pikpak-use-trash=false` to delete files permanently instead.",
+			Advanced: true,
+		}, {
+			Name:     "trashed_only",
+			Default:  false,
+			Help:     "Only show files that are in the trash.\n\nThis will show trashed files in their original directory structure.",
+			Advanced: true,
+		}, {
+			Name:     "hash_memory_limit",
+			Help:     "Files bigger than this will be cached on disk to calculate hash if required.",
+			Default:  fs.SizeSuffix(10 * 1024 * 1024),
+			Advanced: true,
+		}, {
+			Name:     config.ConfigEncoding,
+			Help:     config.ConfigEncodingHelp,
+			Advanced: true,
+			Default: (encoder.EncodeCtl |
```

Comment:
How sure are you of these encodings?

Reply:
The `.csv` is the easiest thing to look at here. Look at the `Get L/M/R` and `List L/M/R` colums. This is Get a file with the character on the Left/Middle/Right or List it. If any of these fails it needs some kind of escaping. Some things can be escaped for only use on Left/Right.

Looks like you need to encode
- control chars
- " * : < > ? \ |
- InvalidUtf8
- LeftSpace, RightSpace, RightPeriod (LeftPeriod is OK I think)

I think you can lose a few things from your encoding list as `EncodeWin` covers some of the other ones. I recommend you don't use `EncodeWin` as this isn't a windows platform I don't think.

Note that EncodeLeftCrLfHtVt and EncodeRightCrLfHtVt should be covered by EncodeCtl

`~` seemed to be OK so you don't need EncodeLeftTilde